### PR TITLE
[JENKINS-42658] - Fix handling of offline agents in SlaveComputer methods

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This is only a partial fix, because all the methods are not tolerant against race conditions with agent disconnection. But better than nothing.

No tests, because the changes are trivial.

See [JENKINS-42658](https://issues.jenkins-ci.org/browse/JENKINS-42658).

### Changelog entries

Proposed changelog entries:

* 42658: Internal API: Prevent `NullPointerException` when accessing agent computer information when the agent is offline 
* ...

### Submitter checklist

- [x] JIRA issue is well described
- [x] Link to JIRA ticket in description, if appropriate
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For new API and extension points: Link to the reference implementation in open-source (or example in Javadoc)

### Desired reviewers

@mention

### CODE REVIEW

File: core/src/main/java/hudson/slaves/SlaveComputer.java [hunk 1]
```diff
@@ -749,27 +783,64 @@ protected ComputerLauncher grabLauncher(Node node) {
...
      */
+    @Nonnull
     public String getSlaveVersion() throws IOException, InterruptedException {
+        if (channel == null) {
+            return "Unknown (agent is offline)";
+        }
         return channel.call(new SlaveVersion());
     }
+    
+    //TODO: move to public API and deprecate the old one after the merge to master
```
Comment (daniel-beck) [inline core/src/main/java/hudson/slaves/SlaveComputer.java]: What motivates this change?

Comment (oleg-nenashev) [inline core/src/main/java/hudson/slaves/SlaveComputer.java]: I want to switch to the `CheckForNull` API, and it is a good opportunity to get rid of the "Slave" in API definition. Can remove it from this commit if you prefer

Comment (daniel-beck) [inline core/src/main/java/hudson/slaves/SlaveComputer.java]: We're never going to get rid of `Slave` in relevant APIs, so the attempt just confuses things further.

Comment (oleg-nenashev) [inline core/src/main/java/hudson/slaves/SlaveComputer.java]: OK, will remove

**Comment by jglick** [comment]
Can you bring this up to date? I just saw

```
java.lang.NullPointerException
	at hudson.slaves.SlaveComputer.getClassLoadingTime(SlaveComputer.java:452)
```

from `…/computer/…/systemInfo`.

**Comment by oleg-nenashev** [comment]
OK, will refresh once I get some time for that

**Comment by daniel-beck** [comment]
Closing after a year with no progress or comments. You know how to reopen or just resubmit PRs 😉

**Pull request closed without merging**